### PR TITLE
Adds base functionality for searching and adding Mutations and Diseases to Query 

### PR DESCRIPTION
**Pull request by bdolly**
@dhimmel  @txnnxr @cgreene 
functional prototype (best viewed in Chrome): https://bdolly.github.io/cognoma-frontend/#/query-builder/mutations

This adds the basic minimum viable functionality of searching for Mutations and Disease Types and adding them to the main query. 

@awm33 It loads the Genes/Mutations and Diseases from an angular mock backend implementation modeled after the proposed [core-service](https://github.com/cognoma/core-service/blob/master/doc/api.md) endpoints and responses. All data is mocked and for testing purposes only. Genes,  Disease, and Samples are pulled from static json files in the build and Disease Sample Mutations are randomly generated for mock purposes. 

Disease positives and negatives counts are update in real-time as mutations are added and removed from the query

### CODE REVIEW

**Comment by txnnxr** [comment]
@bdolly this is awesome, just took a quick look only things that seems off immediately 
- info, X, and sort icons  aren't showing up 
also, I won't be at the meeting tonight but just as an update: I'm writing as many unit tests I can think of, and I was thinking I might tackle the Query Bar at the bottom of the app next. 
Let me know what you think.

**Comment by dhimmel** [comment]
Yes this is terrific!.

> I won't be at the meeting tonight

@txnnxr no meeting tonight. Next meeting on [December 6](https://www.meetup.com/DataPhilly/events/235882962/).

**Comment by bdolly** [comment]
@txnnxr 

> info, X, and sort icons aren't showing up

This is an `https` issue with bootstrap and git hub pages hosting that I haven't worked out yet 

> I'm writing as many unit tests I can think of

I wrote a bunch of unit test for the existing components in `test/unit`, feel free to reference/change/update those as you see fit. There's also some issues with the search results sorting if you want to take a stab at those. 

Another set of tasks in my sights soon is to tidy up and lay some good css foundations and variables. I know it's all early stages right now but a good css architecture will help keep things tidy moving forward. 

I'm going to try and organize the front-end github project kanban board in the next couple days so it's up-to-date with the project status and hopefully see if I can recruit some more front-end devs to help us out so we can get an minimum viable product up soon.

**Comment by cgreene** [comment]
@bdolly this is spiffy! Does a protocol relative URL work to avoid the insecure scripts issue ( https://www.paulirish.com/2010/the-protocol-relative-url/ )?

**Comment by dhimmel** [comment]
Doesn't have to be in this pull request, but it would be great to have a "Add all cancers" button on the DISEASE TYPE page. I imagine many users will be interested in fitting models on all samples.

**Comment by bdolly** [comment]
@dhimmel so that would add all 33 Disease Types to the query?

**Comment by dhimmel** [comment]
> @dhimmel so that would add all 33 Disease Types to the query?

Exactly

File: app/js/utils/DateTime.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/**
```
Comment (awm33) [inline app/js/utils/DateTime.js]: Have you locked at using something like [moment.js](http://momentjs.com/) instead of rolling you own?

**Comment by awm33** [comment]
Some functional testing notes:

- Copy-pasting a gene symbol does not work. I had to backspace a character. It also looks like hitting enter did not help.
- Drag and drop doesn't work for me, but clicking the gene will add it to the list.
- When adding something by click, the UI doesn't really hint that that will happen. It keeps the cursor mouse icon and nothing visual suggests that the gene or disease card is a button. I clicked a card by accident.
- The squares everywhere, which I thing are just a result of glyphicons not loading. Looks like bootstrap.min.css is getting an https/http error. You can avoid that by doing `<link href="//netdna.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet">` then it will just use the protocol used to load the page.

Attaching a screenshot of what it looks like in my browser.

screen shot 2016-12-03 at 7 03 44 pm

File: app/js/utils/DateTime.js (hunk 1 above)
Comment (bdolly) [inline app/js/utils/DateTime.js]: I grabbed the LogDecorator from a tutorial I found. I think moment.js is a bit much for what the DateTime is used for in the LogDecorator. If it was part of the UI or essential business logic I would add moment.js as a dependency.

**Pull request merged**